product_packaging_container_deposit/models/container_deposit_order_line_mixin.py

```diff
@@ -0,0 +1,59 @@
+# Copyright 2023 Camptocamp (<https://www.camptocamp.com>).
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+
+
+class OrderLineMixin(models.AbstractModel):
+    """This mixin should only be inherited by purchase.order.line and sale.order.line models."""
+
+    _name = "container.deposit.order.line.mixin"
+    _description = "Container Deposit Order Line Mixin"
+
+    is_container_deposit = fields.Boolean()
+
+    def _get_product_qty_field(self):
+        return (
+            self._name == "sale.order.line"
+            and "product_uom_qty"
+            or self._name == "purchase.order.line"
+            and "product_qty"
+        )
```

Comment:
```suggestion
    def _get_product_qty_field(self):
        raise NotImplementedError()
`

and in sale module : 
```
    def _get_product_qty_field(self):
        return "product_uom_qty"
```

and in purchase module : 
```
    def _get_product_qty_field(self):
        return "product_qty"
```

Reply:
oh yeah :)